aptos-move/aptos-sdk-builder/src/rust.rs

```diff
@@ -874,46 +861,18 @@ fn decode_{}_argument(arg: TransactionArgument) -> Option<{}> {{
             U64 => "u64".into(),
             U128 => "u128".into(),
             Address => "AccountAddress".into(),
-            Vector(type_tag) => match type_tag.as_ref() {
-                U8 => {
-                    if local_types {
-                        "Vec<u8>".into()
-                    } else {
-                        "Bytes".into()
-                    }
-                }
-                Bool => "Vec<bool>".into(),
-                U64 => "Vec<u64>".into(),
-                U128 => "Vec<u128>".into(),
-                Address => "Vec<AccountAddress>".into(),
-                Vector(type_tag) if type_tag.as_ref() == &U8 => "Vec<Vec<u8>>".into(),
+            Vector(type_tag) => {
+                format!("Vec<{}>", Self::quote_type(type_tag.as_ref(), local_types))
+            }
```

Comment:
this might not be true, I don't know if vec<vec<vec<t>>> is supported, is it?

Reply:
This is illegal in abi.  The SDK builder won't have to deal with `Vec<Vec<....>>` txn args.

Reply:
Reading these comments, I was confused. 
abi does support vec embedding more than twice. I was able to generate ABI for
```
    public entry fun a_test(arg: vector<vector<vector<address>>>) {
        let _ = arg;
    }
```

However, the limitation is on the SDK generator side. Before this PR, we only support `vec<vec<u8>>`. This PR generalizes vector embedding and removed the constraint of just `vec<vec<u8>>`.  The question is do we want to make the generalization? This generalization would also require SDK updates and be consistent for script arguments https://github.com/aptos-labs/aptos-core/pull/2740/files#diff-5f7dcf17c3c592d2b126982779cc63893f012b913cdf583f8188f154bff23cf5R887-R900. 

I am conservative on generalizing this.

Reply:
Conceptually, transaction arguments should have always allowed arbitrary nesting. This is BTW also how validator in AptosVM works. Let's go forward with this instead of building awkward ad-hoc restrictions. If there is any upstream code (TS, CLI, ..) which can't deal with this it should bail out and we can fix it there. I expect generalizing makes things only easier, as this PR shows.